diff -ur krita-4.4.0/libs/brush/CMakeLists.txt a/libs/brush/CMakeLists.txt
--- krita-4.4.0/libs/brush/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/libs/brush/CMakeLists.txt	2020-10-26 20:20:27.480092525 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory( tests )
+endif()
 
 include_directories(SYSTEM
     ${EIGEN3_INCLUDE_DIR}
diff -ur krita-4.4.0/libs/flake/CMakeLists.txt a/libs/flake/CMakeLists.txt
--- krita-4.4.0/libs/flake/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/libs/flake/CMakeLists.txt	2020-10-26 20:20:27.479092535 +0100
@@ -9,8 +9,12 @@
 )
 
 add_subdirectory(styles)
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
+if(BUILD_TESTING)
 add_subdirectory(resources/tests)
+endif()
 
 set(kritaflake_SRCS
     KoGradientHelper.cpp
diff -ur krita-4.4.0/libs/flake/resources/tests/CMakeLists.txt a/libs/flake/resources/tests/CMakeLists.txt
--- krita-4.4.0/libs/flake/resources/tests/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/libs/flake/resources/tests/CMakeLists.txt	2020-10-26 20:20:27.479092535 +0100
@@ -2,7 +2,9 @@
 
 include(ECMAddTests)
 
+if(BUILD_TESTING)
 include_directories(${CMAKE_SOURCE_DIR}/sdk/tests)
+endif()
 
 macro_add_unittest_definitions()
 
diff -ur krita-4.4.0/libs/global/CMakeLists.txt a/libs/global/CMakeLists.txt
--- krita-4.4.0/libs/global/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/libs/global/CMakeLists.txt	2020-10-26 20:20:27.487092454 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory( tests )
+endif()
 
 include(CheckFunctionExists)
 check_function_exists(backtrace HAVE_BACKTRACE)
diff -ur krita-4.4.0/libs/image/CMakeLists.txt a/libs/image/CMakeLists.txt
--- krita-4.4.0/libs/image/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/libs/image/CMakeLists.txt	2020-10-26 20:20:27.484092484 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory( tests )
+endif()
 
 add_subdirectory( tiles3 )
 
diff -ur krita-4.4.0/libs/image/tiles3/CMakeLists.txt a/libs/image/tiles3/CMakeLists.txt
--- krita-4.4.0/libs/image/tiles3/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/libs/image/tiles3/CMakeLists.txt	2020-10-26 20:20:27.485092474 +0100
@@ -1 +1,3 @@
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
diff -ur krita-4.4.0/libs/libkis/CMakeLists.txt a/libs/libkis/CMakeLists.txt
--- krita-4.4.0/libs/libkis/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/libs/libkis/CMakeLists.txt	2020-10-26 20:20:27.482092504 +0100
@@ -48,4 +48,6 @@
 install(TARGETS kritalibkis  ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
diff -ur krita-4.4.0/libs/metadata/CMakeLists.txt a/libs/metadata/CMakeLists.txt
--- krita-4.4.0/libs/metadata/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/libs/metadata/CMakeLists.txt	2020-10-26 20:20:27.477092555 +0100
@@ -29,4 +29,6 @@
 
 install(TARGETS kritametadata  ${INSTALL_TARGETS_DEFAULT_ARGS})
 
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
diff -ur krita-4.4.0/libs/odf/CMakeLists.txt a/libs/odf/CMakeLists.txt
--- krita-4.4.0/libs/odf/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/libs/odf/CMakeLists.txt	2020-10-26 20:20:27.491092413 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory( tests )
+endif()
 
 set(kritaodf_LIB_SRCS
     KoOdf.cpp
diff -ur krita-4.4.0/libs/pigment/benchmarks/CMakeLists.txt a/libs/pigment/benchmarks/CMakeLists.txt
--- krita-4.4.0/libs/pigment/benchmarks/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/libs/pigment/benchmarks/CMakeLists.txt	2020-10-26 20:46:07.438753996 +0100
@@ -4,10 +4,10 @@
 ########### next target ###############
 
 set(ko_colorspaces_benchmark_SRCS KoColorSpacesBenchmark.cpp)
-krita_add_benchmark(KoColorSpacesBenchmark TESTNAME pigment-benchmarks-KoColorSpacesBenchmark ${ko_colorspaces_benchmark_SRCS})
-target_link_libraries(KoColorSpacesBenchmark kritapigment KF5::I18n  Qt5::Test)
+#krita_add_benchmark(KoColorSpacesBenchmark TESTNAME pigment-benchmarks-KoColorSpacesBenchmark ${ko_colorspaces_benchmark_SRCS})
+#target_link_libraries(KoColorSpacesBenchmark kritapigment KF5::I18n )
 
 set(ko_compositeops_benchmark_SRCS KoCompositeOpsBenchmark.cpp)
-krita_add_benchmark(KoCompositeOpsBenchmark TESTNAME pigment-benchmarks-KoCompositeOpsBenchmark ${ko_compositeops_benchmark_SRCS})
-target_link_libraries(KoCompositeOpsBenchmark  kritapigment KF5::I18n  Qt5::Test)
+#krita_add_benchmark(KoCompositeOpsBenchmark TESTNAME pigment-benchmarks-KoCompositeOpsBenchmark ${ko_compositeops_benchmark_SRCS})
+#target_link_libraries(KoCompositeOpsBenchmark  kritapigment KF5::I18n )
 
diff -ur krita-4.4.0/libs/pigment/CMakeLists.txt a/libs/pigment/CMakeLists.txt
--- krita-4.4.0/libs/pigment/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/libs/pigment/CMakeLists.txt	2020-10-26 20:20:27.477092555 +0100
@@ -33,7 +33,9 @@
     set(__per_arch_alpha_applicator_factory_objs KoAlphaMaskApplicatorFactoryImpl.cpp)
 endif()
 
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
 add_subdirectory(benchmarks)
 
 set(kritapigment_SRCS
diff -ur krita-4.4.0/libs/store/CMakeLists.txt a/libs/store/CMakeLists.txt
--- krita-4.4.0/libs/store/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/libs/store/CMakeLists.txt	2020-10-26 20:20:27.482092504 +0100
@@ -1,6 +1,8 @@
 include_directories(${QUAZIP_INCLUDE_DIRS})
 
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
 
 set(kritastore_LIB_SRCS
     KoDirectoryStore.cpp
diff -ur krita-4.4.0/libs/ui/CMakeLists.txt a/libs/ui/CMakeLists.txt
--- krita-4.4.0/libs/ui/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/libs/ui/CMakeLists.txt	2020-10-26 20:20:27.490092423 +0100
@@ -13,7 +13,9 @@
     include_directories (${Qt5AndroidExtras_INCLUDE_DIRS})
 endif()
 
+if(BUILD_TESTING)
 add_subdirectory( tests )
+endif()
 
 if (APPLE)
     find_library(FOUNDATION_LIBRARY Foundation)
diff -ur krita-4.4.0/libs/ui/tests/CMakeLists.txt a/libs/ui/tests/CMakeLists.txt
--- krita-4.4.0/libs/ui/tests/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/libs/ui/tests/CMakeLists.txt	2020-10-26 20:20:27.491092413 +0100
@@ -1,7 +1,9 @@
 set( EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR} )
 
 include_directories(${CMAKE_SOURCE_DIR}/libs/image/metadata
+if(BUILD_TESTING)
                     ${CMAKE_SOURCE_DIR}/sdk/tests )
+endif()
 
 include(ECMAddTests)
 
diff -ur krita-4.4.0/libs/widgets/CMakeLists.txt a/libs/widgets/CMakeLists.txt
--- krita-4.4.0/libs/widgets/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/libs/widgets/CMakeLists.txt	2020-10-26 20:20:27.486092464 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory( tests )
+endif()
 
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
diff -ur krita-4.4.0/libs/widgetutils/CMakeLists.txt a/libs/widgetutils/CMakeLists.txt
--- krita-4.4.0/libs/widgetutils/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/libs/widgetutils/CMakeLists.txt	2020-10-26 20:20:27.483092494 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
 
 configure_file(xmlgui/config-xmlgui.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config-xmlgui.h )
 
diff -ur krita-4.4.0/plugins/color/lcms2engine/CMakeLists.txt a/plugins/color/lcms2engine/CMakeLists.txt
--- krita-4.4.0/plugins/color/lcms2engine/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/color/lcms2engine/CMakeLists.txt	2020-10-26 20:20:27.520092118 +0100
@@ -1,6 +1,8 @@
 project( lcmsengine )
 
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
 
 include_directories(SYSTEM
     ${LCMS2_INCLUDE_DIR}
diff -ur krita-4.4.0/plugins/dockers/animation/CMakeLists.txt a/plugins/dockers/animation/CMakeLists.txt
--- krita-4.4.0/plugins/dockers/animation/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/dockers/animation/CMakeLists.txt	2020-10-26 20:20:27.506092260 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
 
 set(KRITA_ANIMATIONDOCKER_SOURCES
     animation_dockers.cpp
diff -ur krita-4.4.0/plugins/dockers/layerdocker/CMakeLists.txt a/plugins/dockers/layerdocker/CMakeLists.txt
--- krita-4.4.0/plugins/dockers/layerdocker/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/dockers/layerdocker/CMakeLists.txt	2020-10-26 20:20:27.505092271 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 #add_subdirectory(tests)
+endif()
 
 set(kritalayerdocker_SOURCES 
     LayerBox.cpp
diff -ur krita-4.4.0/plugins/dockers/layerdocker/tests/CMakeLists.txt a/plugins/dockers/layerdocker/tests/CMakeLists.txt
--- krita-4.4.0/plugins/dockers/layerdocker/tests/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/dockers/layerdocker/tests/CMakeLists.txt	2020-10-26 20:20:27.505092271 +0100
@@ -1,7 +1,9 @@
 set( EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR} )
   
 include_directories(${CMAKE_SOURCE_DIR}/libs/image/metadata
+if(BUILD_TESTING)
                     ${CMAKE_SOURCE_DIR}/sdk/tests )
+endif()
 include(ECMAddTests)
 
 macro_add_unittest_definitions()
diff -ur krita-4.4.0/plugins/dockers/lut/CMakeLists.txt a/plugins/dockers/lut/CMakeLists.txt
--- krita-4.4.0/plugins/dockers/lut/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/dockers/lut/CMakeLists.txt	2020-10-26 20:20:27.504092281 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
 
 include_directories(SYSTEM
     ${OCIO_INCLUDE_DIR}
diff -ur krita-4.4.0/plugins/extensions/qmic/CMakeLists.txt a/plugins/extensions/qmic/CMakeLists.txt
--- krita-4.4.0/plugins/extensions/qmic/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/extensions/qmic/CMakeLists.txt	2020-10-26 20:20:27.530092016 +0100
@@ -20,4 +20,6 @@
 target_link_libraries(kritaqmic kritaui)
 install(TARGETS kritaqmic  DESTINATION ${KRITA_PLUGIN_INSTALL_DIR})
 
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
diff -ur krita-4.4.0/plugins/filters/CMakeLists.txt a/plugins/filters/CMakeLists.txt
--- krita-4.4.0/plugins/filters/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/filters/CMakeLists.txt	2020-10-26 20:20:27.512092199 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory( tests )
+endif()
 add_subdirectory( blur )
 add_subdirectory( colors )
 add_subdirectory( colorsfilters )
diff -ur krita-4.4.0/plugins/filters/unsharp/CMakeLists.txt a/plugins/filters/unsharp/CMakeLists.txt
--- krita-4.4.0/plugins/filters/unsharp/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/filters/unsharp/CMakeLists.txt	2020-10-26 20:20:27.515092169 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory( tests )
+endif()
 
 set(kritaunsharpfilter_SOURCES
     unsharp.cpp
diff -ur krita-4.4.0/plugins/filters/unsharp/tests/CMakeLists.txt a/plugins/filters/unsharp/tests/CMakeLists.txt
--- krita-4.4.0/plugins/filters/unsharp/tests/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/filters/unsharp/tests/CMakeLists.txt	2020-10-26 20:20:27.515092169 +0100
@@ -1,5 +1,7 @@
 set( EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR} )
+if(BUILD_TESTING)
 include_directories( ${CMAKE_SOURCE_DIR}/sdk/tests )
+endif()
 
 macro_add_unittest_definitions()
 
diff -ur krita-4.4.0/plugins/generators/screentone/CMakeLists.txt a/plugins/generators/screentone/CMakeLists.txt
--- krita-4.4.0/plugins/generators/screentone/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/generators/screentone/CMakeLists.txt	2020-10-26 20:20:27.511092210 +0100
@@ -12,4 +12,6 @@
 target_link_libraries(kritascreentonegenerator kritaui)
 install(TARGETS kritascreentonegenerator DESTINATION ${KRITA_PLUGIN_INSTALL_DIR})
 
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
diff -ur krita-4.4.0/plugins/generators/seexpr/CMakeLists.txt a/plugins/generators/seexpr/CMakeLists.txt
--- krita-4.4.0/plugins/generators/seexpr/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/generators/seexpr/CMakeLists.txt	2020-10-26 20:20:27.511092210 +0100
@@ -18,4 +18,6 @@
 
 install(TARGETS kritaseexprgenerator  DESTINATION ${KRITA_PLUGIN_INSTALL_DIR})
 
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
diff -ur krita-4.4.0/plugins/impex/brush/CMakeLists.txt a/plugins/impex/brush/CMakeLists.txt
--- krita-4.4.0/plugins/impex/brush/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/brush/CMakeLists.txt	2020-10-26 20:20:27.525092067 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
 
 set(kritabrushexport_PART_SRCS
     kis_brush_export.cpp
diff -ur krita-4.4.0/plugins/impex/brush/tests/CMakeLists.txt a/plugins/impex/brush/tests/CMakeLists.txt
--- krita-4.4.0/plugins/impex/brush/tests/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/brush/tests/CMakeLists.txt	2020-10-26 20:20:27.526092057 +0100
@@ -1,5 +1,7 @@
 set( EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR} )
+if(BUILD_TESTING)
 include_directories(     ${CMAKE_SOURCE_DIR}/sdk/tests )
+endif()
 
 include(KritaAddBrokenUnitTest)
 
diff -ur krita-4.4.0/plugins/impex/csv/CMakeLists.txt a/plugins/impex/csv/CMakeLists.txt
--- krita-4.4.0/plugins/impex/csv/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/csv/CMakeLists.txt	2020-10-26 20:20:27.524092077 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
 
 # import
 
diff -ur krita-4.4.0/plugins/impex/csv/tests/CMakeLists.txt a/plugins/impex/csv/tests/CMakeLists.txt
--- krita-4.4.0/plugins/impex/csv/tests/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/csv/tests/CMakeLists.txt	2020-10-26 20:20:27.524092077 +0100
@@ -1,5 +1,7 @@
 set( EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR} )
+if(BUILD_TESTING)
 include_directories(     ${CMAKE_SOURCE_DIR}/sdk/tests )
+endif()
 
 include(KritaAddBrokenUnitTest)
 
diff -ur krita-4.4.0/plugins/impex/exr/CMakeLists.txt a/plugins/impex/exr/CMakeLists.txt
--- krita-4.4.0/plugins/impex/exr/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/exr/CMakeLists.txt	2020-10-26 20:20:27.522092098 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
 
 include_directories(SYSTEM ${OPENEXR_INCLUDE_DIRS} )
 
diff -ur krita-4.4.0/plugins/impex/exr/tests/CMakeLists.txt a/plugins/impex/exr/tests/CMakeLists.txt
--- krita-4.4.0/plugins/impex/exr/tests/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/exr/tests/CMakeLists.txt	2020-10-26 20:20:27.522092098 +0100
@@ -1,5 +1,7 @@
 set( EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR} )
+if(BUILD_TESTING)
 include_directories(     ${CMAKE_SOURCE_DIR}/sdk/tests )
+endif()
 
 include(KritaAddBrokenUnitTest)
 
diff -ur krita-4.4.0/plugins/impex/gif/CMakeLists.txt a/plugins/impex/gif/CMakeLists.txt
--- krita-4.4.0/plugins/impex/gif/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/gif/CMakeLists.txt	2020-10-26 20:20:27.524092077 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
 
 set(kritagifexport_SOURCES
     kis_gif_export.cpp
diff -ur krita-4.4.0/plugins/impex/gif/tests/CMakeLists.txt a/plugins/impex/gif/tests/CMakeLists.txt
--- krita-4.4.0/plugins/impex/gif/tests/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/gif/tests/CMakeLists.txt	2020-10-26 20:20:27.524092077 +0100
@@ -1,5 +1,7 @@
 set( EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR} )
+if(BUILD_TESTING)
 include_directories(     ${CMAKE_SOURCE_DIR}/sdk/tests )
+endif()
 
 include(KritaAddBrokenUnitTest)
 
diff -ur krita-4.4.0/plugins/impex/heif/CMakeLists.txt a/plugins/impex/heif/CMakeLists.txt
--- krita-4.4.0/plugins/impex/heif/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/heif/CMakeLists.txt	2020-10-26 20:20:27.524092077 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
 
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${HEIF_CFLAGS}")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${HEIF_CFLAGS}")
diff -ur krita-4.4.0/plugins/impex/heif/tests/CMakeLists.txt a/plugins/impex/heif/tests/CMakeLists.txt
--- krita-4.4.0/plugins/impex/heif/tests/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/heif/tests/CMakeLists.txt	2020-10-26 20:20:27.524092077 +0100
@@ -1,5 +1,7 @@
 set( EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR} )
+if(BUILD_TESTING)
 include_directories(     ${CMAKE_SOURCE_DIR}/sdk/tests )
+endif()
 
 include(KritaAddBrokenUnitTest)
 
diff -ur krita-4.4.0/plugins/impex/heightmap/CMakeLists.txt a/plugins/impex/heightmap/CMakeLists.txt
--- krita-4.4.0/plugins/impex/heightmap/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/heightmap/CMakeLists.txt	2020-10-26 20:20:27.526092057 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
 
 include_directories( ${CMAKE_CURRENT_SOURCE_DIR} )
 
diff -ur krita-4.4.0/plugins/impex/heightmap/tests/CMakeLists.txt a/plugins/impex/heightmap/tests/CMakeLists.txt
--- krita-4.4.0/plugins/impex/heightmap/tests/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/heightmap/tests/CMakeLists.txt	2020-10-26 20:20:27.526092057 +0100
@@ -1,5 +1,7 @@
 set( EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR} )
+if(BUILD_TESTING)
 include_directories(     ${CMAKE_SOURCE_DIR}/sdk/tests )
+endif()
 
 include(KritaAddBrokenUnitTest)
 
diff -ur krita-4.4.0/plugins/impex/jp2/CMakeLists.txt a/plugins/impex/jp2/CMakeLists.txt
--- krita-4.4.0/plugins/impex/jp2/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/jp2/CMakeLists.txt	2020-10-26 20:20:27.521092108 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
 
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenJPEG_CFLAGS}")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenJPEG_CFLAGS}")
diff -ur krita-4.4.0/plugins/impex/jp2/tests/CMakeLists.txt a/plugins/impex/jp2/tests/CMakeLists.txt
--- krita-4.4.0/plugins/impex/jp2/tests/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/jp2/tests/CMakeLists.txt	2020-10-26 20:20:27.521092108 +0100
@@ -1,5 +1,7 @@
 set( EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR} )
+if(BUILD_TESTING)
 include_directories(     ${CMAKE_SOURCE_DIR}/sdk/tests )
+endif()
 
 include(KritaAddBrokenUnitTest)
 
diff -ur krita-4.4.0/plugins/impex/jpeg/CMakeLists.txt a/plugins/impex/jpeg/CMakeLists.txt
--- krita-4.4.0/plugins/impex/jpeg/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/jpeg/CMakeLists.txt	2020-10-26 20:20:27.523092088 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
 
 set(ICCJPEG_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/lcms")
 
diff -ur krita-4.4.0/plugins/impex/jpeg/tests/CMakeLists.txt a/plugins/impex/jpeg/tests/CMakeLists.txt
--- krita-4.4.0/plugins/impex/jpeg/tests/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/jpeg/tests/CMakeLists.txt	2020-10-26 20:20:27.523092088 +0100
@@ -1,5 +1,7 @@
 #set EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR})
+if(BUILD_TESTING)
 include_directories(${CMAKE_SOURCE_DIR}/sdk/tests)
+endif()
 include(KritaAddBrokenUnitTest)
 
 macro_add_unittest_definitions()
diff -ur krita-4.4.0/plugins/impex/libkra/CMakeLists.txt a/plugins/impex/libkra/CMakeLists.txt
--- krita-4.4.0/plugins/impex/libkra/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/libkra/CMakeLists.txt	2020-10-26 20:20:27.527092047 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
 
 set(kritalibkra_LIB_SRCS
     kis_colorize_dom_utils.cpp
diff -ur krita-4.4.0/plugins/impex/libkra/tests/CMakeLists.txt a/plugins/impex/libkra/tests/CMakeLists.txt
--- krita-4.4.0/plugins/impex/libkra/tests/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/libkra/tests/CMakeLists.txt	2020-10-26 20:20:27.527092047 +0100
@@ -1,6 +1,8 @@
 set( EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR} )
 
+if(BUILD_TESTING)
 include_directories( ${CMAKE_SOURCE_DIR}/sdk/tests )
+endif()
 
 macro_add_unittest_definitions()
 macro_add_unittest_definitions()
diff -ur krita-4.4.0/plugins/impex/ora/CMakeLists.txt a/plugins/impex/ora/CMakeLists.txt
--- krita-4.4.0/plugins/impex/ora/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/ora/CMakeLists.txt	2020-10-26 20:20:27.521092108 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
 
 set(libkritaconverter_LIB_SRCS
     ora_converter.cpp
diff -ur krita-4.4.0/plugins/impex/ora/tests/CMakeLists.txt a/plugins/impex/ora/tests/CMakeLists.txt
--- krita-4.4.0/plugins/impex/ora/tests/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/ora/tests/CMakeLists.txt	2020-10-26 20:20:27.521092108 +0100
@@ -1,5 +1,7 @@
 set( EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR} )
+if(BUILD_TESTING)
 include_directories(     ${CMAKE_SOURCE_DIR}/sdk/tests )
+endif()
 
 include(KritaAddBrokenUnitTest)
 
diff -ur krita-4.4.0/plugins/impex/pdf/CMakeLists.txt a/plugins/impex/pdf/CMakeLists.txt
--- krita-4.4.0/plugins/impex/pdf/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/pdf/CMakeLists.txt	2020-10-26 20:20:27.522092098 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
 
 set(kritapdfimport_SOURCES kis_pdf_import.cpp kis_pdf_import_widget.cpp )
 
diff -ur krita-4.4.0/plugins/impex/pdf/tests/CMakeLists.txt a/plugins/impex/pdf/tests/CMakeLists.txt
--- krita-4.4.0/plugins/impex/pdf/tests/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/pdf/tests/CMakeLists.txt	2020-10-26 20:20:27.523092088 +0100
@@ -1,5 +1,7 @@
 set( EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR} )
+if(BUILD_TESTING)
 include_directories(     ${CMAKE_SOURCE_DIR}/sdk/tests )
+endif()
 
 include(KritaAddBrokenUnitTest)
 
diff -ur krita-4.4.0/plugins/impex/png/CMakeLists.txt a/plugins/impex/png/CMakeLists.txt
--- krita-4.4.0/plugins/impex/png/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/png/CMakeLists.txt	2020-10-26 20:20:27.521092108 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
 
 set(kritapngimport_SOURCES
     kis_png_import.cc
diff -ur krita-4.4.0/plugins/impex/png/tests/CMakeLists.txt a/plugins/impex/png/tests/CMakeLists.txt
--- krita-4.4.0/plugins/impex/png/tests/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/png/tests/CMakeLists.txt	2020-10-26 20:20:27.522092098 +0100
@@ -1,5 +1,7 @@
 set( EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR} )
+if(BUILD_TESTING)
 include_directories(     ${CMAKE_SOURCE_DIR}/sdk/tests )
+endif()
 
 include(KritaAddBrokenUnitTest)
 
diff -ur krita-4.4.0/plugins/impex/psd/CMakeLists.txt a/plugins/impex/psd/CMakeLists.txt
--- krita-4.4.0/plugins/impex/psd/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/psd/CMakeLists.txt	2020-10-26 20:20:27.525092067 +0100
@@ -1,6 +1,8 @@
 if (NOT MSVC AND NOT APPLE)
+if(BUILD_TESTING)
     add_subdirectory(tests)
 endif()
+endif()
 
 configure_file(config_psd.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config_psd.h)
 
diff -ur krita-4.4.0/plugins/impex/qimageio/CMakeLists.txt a/plugins/impex/qimageio/CMakeLists.txt
--- krita-4.4.0/plugins/impex/qimageio/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/qimageio/CMakeLists.txt	2020-10-26 20:20:27.522092098 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
 
 set(kritaqimageioexport_SOURCES
     kis_qimageio_export.cpp
diff -ur krita-4.4.0/plugins/impex/qimageio/tests/CMakeLists.txt a/plugins/impex/qimageio/tests/CMakeLists.txt
--- krita-4.4.0/plugins/impex/qimageio/tests/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/qimageio/tests/CMakeLists.txt	2020-10-26 20:20:27.522092098 +0100
@@ -1,5 +1,7 @@
 set( EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR} )
+if(BUILD_TESTING)
 include_directories(     ${CMAKE_SOURCE_DIR}/sdk/tests )
+endif()
 
 include(KritaAddBrokenUnitTest)
 
diff -ur krita-4.4.0/plugins/impex/qml/CMakeLists.txt a/plugins/impex/qml/CMakeLists.txt
--- krita-4.4.0/plugins/impex/qml/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/qml/CMakeLists.txt	2020-10-26 20:20:27.522092098 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
 
 set(kritaqmlexport_SOURCES
     qml_converter.cc
diff -ur krita-4.4.0/plugins/impex/qml/tests/CMakeLists.txt a/plugins/impex/qml/tests/CMakeLists.txt
--- krita-4.4.0/plugins/impex/qml/tests/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/qml/tests/CMakeLists.txt	2020-10-26 20:20:27.522092098 +0100
@@ -1,5 +1,7 @@
 set( EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR} )
+if(BUILD_TESTING)
 include_directories(     ${CMAKE_SOURCE_DIR}/sdk/tests )
+endif()
 
 include(KritaAddBrokenUnitTest)
 
diff -ur krita-4.4.0/plugins/impex/raw/CMakeLists.txt a/plugins/impex/raw/CMakeLists.txt
--- krita-4.4.0/plugins/impex/raw/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/raw/CMakeLists.txt	2020-10-26 20:20:27.526092057 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
 
 if(OPENEXR_FOUND)
 	include_directories(${OPENEXR_INCLUDE_DIRS})
diff -ur krita-4.4.0/plugins/impex/raw/tests/CMakeLists.txt a/plugins/impex/raw/tests/CMakeLists.txt
--- krita-4.4.0/plugins/impex/raw/tests/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/raw/tests/CMakeLists.txt	2020-10-26 20:20:27.526092057 +0100
@@ -1,5 +1,7 @@
 set( EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR} )
+if(BUILD_TESTING)
 include_directories(     ${CMAKE_SOURCE_DIR}/sdk/tests )
+endif()
 
 include(KritaAddBrokenUnitTest)
 
diff -ur krita-4.4.0/plugins/impex/svg/CMakeLists.txt a/plugins/impex/svg/CMakeLists.txt
--- krita-4.4.0/plugins/impex/svg/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/svg/CMakeLists.txt	2020-10-26 20:20:27.523092088 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
 
 set(kritasvgimport_SOURCES
     kis_svg_import.cc
diff -ur krita-4.4.0/plugins/impex/svg/tests/CMakeLists.txt a/plugins/impex/svg/tests/CMakeLists.txt
--- krita-4.4.0/plugins/impex/svg/tests/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/svg/tests/CMakeLists.txt	2020-10-26 20:20:27.523092088 +0100
@@ -1,5 +1,7 @@
 set(EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR})
+if(BUILD_TESTING)
 include_directories(${CMAKE_SOURCE_DIR}/sdk/tests)
+endif()
 include(ECMAddTests)
 macro_add_unittest_definitions()
 include(KritaAddBrokenUnitTest)
diff -ur krita-4.4.0/plugins/impex/tga/CMakeLists.txt a/plugins/impex/tga/CMakeLists.txt
--- krita-4.4.0/plugins/impex/tga/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/tga/CMakeLists.txt	2020-10-26 20:20:27.521092108 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
 
 set(kritatgaexport_SOURCES
     kis_tga_export.cpp
diff -ur krita-4.4.0/plugins/impex/tga/tests/CMakeLists.txt a/plugins/impex/tga/tests/CMakeLists.txt
--- krita-4.4.0/plugins/impex/tga/tests/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/tga/tests/CMakeLists.txt	2020-10-26 20:20:27.521092108 +0100
@@ -1,5 +1,7 @@
 set( EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR} )
+if(BUILD_TESTING)
 include_directories(     ${CMAKE_SOURCE_DIR}/sdk/tests )
+endif()
 
 include(KritaAddBrokenUnitTest)
 
diff -ur krita-4.4.0/plugins/impex/tiff/CMakeLists.txt a/plugins/impex/tiff/CMakeLists.txt
--- krita-4.4.0/plugins/impex/tiff/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/tiff/CMakeLists.txt	2020-10-26 20:20:27.525092067 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
 
 set(libkritatiffconverter_LIB_SRCS
     kis_tiff_converter.cc
diff -ur krita-4.4.0/plugins/impex/tiff/tests/CMakeLists.txt a/plugins/impex/tiff/tests/CMakeLists.txt
--- krita-4.4.0/plugins/impex/tiff/tests/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/tiff/tests/CMakeLists.txt	2020-10-26 20:20:27.525092067 +0100
@@ -3,7 +3,9 @@
 endif()
 
 set( EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR} )
+if(BUILD_TESTING)
 include_directories( ${CMAKE_SOURCE_DIR}/sdk/tests )
+endif()
 
 include(ECMAddTests)
 include(KritaAddBrokenUnitTest)
diff -ur krita-4.4.0/plugins/impex/xcf/CMakeLists.txt a/plugins/impex/xcf/CMakeLists.txt
--- krita-4.4.0/plugins/impex/xcf/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/xcf/CMakeLists.txt	2020-10-26 20:20:27.523092088 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
 
 set(XCFTOOLS_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/xcftools")
 
diff -ur krita-4.4.0/plugins/impex/xcf/tests/CMakeLists.txt a/plugins/impex/xcf/tests/CMakeLists.txt
--- krita-4.4.0/plugins/impex/xcf/tests/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/impex/xcf/tests/CMakeLists.txt	2020-10-26 20:20:27.523092088 +0100
@@ -1,5 +1,7 @@
 set( EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR} )
+if(BUILD_TESTING)
 include_directories(     ${CMAKE_SOURCE_DIR}/sdk/tests )
+endif()
 
 include(KritaAddBrokenUnitTest)
 
diff -ur krita-4.4.0/plugins/paintops/defaultpaintops/brush/tests/CMakeLists.txt a/plugins/paintops/defaultpaintops/brush/tests/CMakeLists.txt
--- krita-4.4.0/plugins/paintops/defaultpaintops/brush/tests/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/paintops/defaultpaintops/brush/tests/CMakeLists.txt	2020-10-26 20:20:27.509092230 +0100
@@ -1,5 +1,7 @@
 set( EXECUTABLE_OUTPUT_PATH ${CMAKE_CURRENT_BINARY_DIR} )
+if(BUILD_TESTING)
 include_directories(     ${CMAKE_SOURCE_DIR}/sdk/tests )
+endif()
 
 include(KritaAddBrokenUnitTest)
 
diff -ur krita-4.4.0/plugins/paintops/defaultpaintops/CMakeLists.txt a/plugins/paintops/defaultpaintops/CMakeLists.txt
--- krita-4.4.0/plugins/paintops/defaultpaintops/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/paintops/defaultpaintops/CMakeLists.txt	2020-10-26 20:20:27.508092240 +0100
@@ -1,4 +1,6 @@
+if(BUILD_TESTING)
 add_subdirectory(brush/tests)
+endif()
 
 include_directories(brush
                     duplicate
diff -ur krita-4.4.0/plugins/paintops/libpaintop/CMakeLists.txt a/plugins/paintops/libpaintop/CMakeLists.txt
--- krita-4.4.0/plugins/paintops/libpaintop/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/paintops/libpaintop/CMakeLists.txt	2020-10-26 20:20:27.509092230 +0100
@@ -106,4 +106,6 @@
 install(TARGETS kritalibpaintop  ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 
+if(BUILD_TESTING)
 add_subdirectory(tests)
+endif()
diff -ur krita-4.4.0/plugins/paintops/libpaintop/tests/CMakeLists.txt a/plugins/paintops/libpaintop/tests/CMakeLists.txt
--- krita-4.4.0/plugins/paintops/libpaintop/tests/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/paintops/libpaintop/tests/CMakeLists.txt	2020-10-26 20:20:27.509092230 +0100
@@ -1,5 +1,7 @@
 ########### next target ###############
+if(BUILD_TESTING)
 include_directories( ${CMAKE_SOURCE_DIR}/..    ${CMAKE_SOURCE_DIR}/sdk/tests )
+endif()
 
 include(KritaAddBrokenUnitTest)
 
diff -ur krita-4.4.0/plugins/tools/basictools/CMakeLists.txt a/plugins/tools/basictools/CMakeLists.txt
--- krita-4.4.0/plugins/tools/basictools/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/tools/basictools/CMakeLists.txt	2020-10-26 20:20:27.518092138 +0100
@@ -1,5 +1,7 @@
 if (NOT APPLE)
+if(BUILD_TESTING)
     add_subdirectory(tests)
+endif()
 endif ()
 
 set(kritadefaulttools_SOURCES
diff -ur krita-4.4.0/plugins/tools/tool_transform2/CMakeLists.txt a/plugins/tools/tool_transform2/CMakeLists.txt
--- krita-4.4.0/plugins/tools/tool_transform2/CMakeLists.txt	2020-10-12 12:10:04.000000000 +0200
+++ a/plugins/tools/tool_transform2/CMakeLists.txt	2020-10-26 20:20:27.519092128 +0100
@@ -1,6 +1,8 @@
 if (NOT WIN32 AND NOT APPLE)
+if(BUILD_TESTING)
     add_subdirectory(tests)
 endif()
+endif()
 
 set(kritatooltransform_SOURCES
     tool_transform.cc
